htdocs/core/modules/export/export_csv.modules.php

```diff
@@ -299,17 +299,26 @@ function close_file()
 	 */
 	function csv_clean($newvalue, $charset)
 	{
+		global $conf;
 		$addquote=0;
+		
 
 		// Rule Dolibarr: No HTML
 		//print $charset.' '.$newvalue."\n";
 		$newvalue=dol_string_nohtmltag($newvalue,1,$charset);
 		//print $charset.' '.$newvalue."\n";
 
 		// Rule 1 CSV: No CR, LF in cells
-		$newvalue=str_replace("\r",'',$newvalue);
+		$oldvalue=$newvalue;
+		$newvalue=str_replace("\r",'\r',$newvalue);
 		$newvalue=str_replace("\n",'\n',$newvalue);
-
+		if ($conf->global->USE_STRICT_CSV_RULES && $oldvalue != $newvalue)
```

Comment:
Can you replace the if ($conf by
if (! empty($conf->

To avoid warning on some php versions ?